modules/rest-api/pages/rest-statistics-single.adoc

```diff
@@ -0,0 +1,487 @@
+
+
+[#curl-syntax]
+== Curl Syntax
+
+----
+curl -X GET -u <username:password> \
+http://<ip-address-or-domain-name>:<port-number>/pools/default/stats/range/<metric-name>/
+  <function-expression>
+----
+
+The `function-expression` must take the following form:
+
+----
+[<function>...]?[<label>=<val>&...]
+----
+
+This syntax is described in the subsections immediately below.
+
+[#function]
+=== `function`
+
+The `function` parameter can be specified as follows:
+
+----
+<function> := rate | irate | increase | avg_over_time | min_over_time | max_over_time | deriv | delta | idelta | sum | min | max | avg | p[1-99]
+----
+
+* `rate`.
+The per-second average rate of increase of the time series in the range vector.
+Breaks in monotonicity (such as counter resets due to target restarts) are automatically adjusted for.
+Also, the calculation extrapolates to the ends of the time range, allowing for missed scrapes or imperfect alignment of scrape cycles with the range's time period.
+
+* `irate`.
+The per-second instant rate of increase of the time series in the range vector.
+This is based on the last two data points. Breaks in monotonicity (such as counter resets due to target restarts) are automatically adjusted for.
+
+* `increase`.
+The increase in the time series in the range vector.
+Breaks in monotonicity (such as counter resets due to target restarts) are automatically adjusted for.
+The increase is extrapolated to cover the full time range as specified in the range vector selector, so that it is possible to get a non-integer result even if a counter increases only by integer increments.
+
+* `avg_over_time`.
+The average value over a specified time window.
+
+* `min_over_time`.
+The minimum value over a specified time window.
+
+* `max_over_time`.
+The maximum value over a specified time window.
+
+* `deriv`.
+The per-second derivative of the time series in a range vector, using simple linear regression.
+
+* `delta`.
+The difference between the first and last value of each time series element in a range vector, returning an instant vector with the given deltas and equivalent labels.
+The delta is extrapolated to cover the full time range as specified in the range vector selector, so that it is possible to get a non-integer result even if the sample values are all integers.
+
+* `idelta`.
+The difference between the last two samples in the range vector, returning an instant vector with the given deltas and equivalent labels.
+
+* `sum`.
+The sum of values retrieved over the sample period.
+
+* `min`.
+The minimum value retrieved over the sample period.
+
+* `max`.
+The maximum value retrieved over the sample period.
+
+* `avg`.
+The average of values retrieved over the sample period.
+
+* `p[1-99]`
+The number of results to be displayed per page.
+
+=== `label` and `value`
+
+The possible specifications of `label`, and their associated possible values, are as follows.
```

Comment:
I think we should probably divide the optional parameters into 2 groups.

(1) the fixed optional parameters: start, end, step, nodes, nodesAggregation, timeWindow, timeout and alighTimestamps.
(2) labels that are in the underlying metrics (such as bucket, scope, collection, ...) -- basically all the labels that show in the metrics reference.

Reply:
Okay I have called them 'General' and 'Couchbase-Specific'. Does that work?